pkg/volume/util.go

```diff
@@ -354,3 +354,251 @@ func UnmountViaEmptyDir(dir string, host VolumeHost, volName string, volSpec Spe
+		if !allowedKeys[expr.Key] {
+			return false, fmt.Errorf("key %q is not permitted in selector.matchExpressions", expr.Key)
+		}
+		if !allowedOperators[expr.Operator] {
+			return false, fmt.Errorf("operator %q is not permitted in selector.matchExpressions", expr.Operator)
+		}
+		if len(expr.Values) < 1 {
+			return false, fmt.Errorf("key %q, operator %q pair does not contain any value(s) in selector.matchExpressions", expr.Key, expr.Operator)
+		}
+	}
+	return false, nil
+}
+
+// GetPVCMatchLabel returns:
+// - either (value, nil) for the key from the matchLabels Selector part of the PVC
+// - or ("", error) in case the key is missing in the matchLabels Selector part of the PVC
+func GetPVCMatchLabel(pvc *v1.PersistentVolumeClaim, key string) (string, error) {
+	if pvc.Spec.Selector == nil {
+		return "", fmt.Errorf("missing selector.matchLabels")
+	}
+	if value, ok := pvc.Spec.Selector.MatchLabels[key]; ok {
+		return value, nil
+	}
+	return "", fmt.Errorf("key %q not found in selector.matchLabels", key)
+}
+
+// GetPVCMatchExpression returns:
+// - either ([]setOfValues, nil) for all matching (key, operator) from the matchExpressions Selector part of the PVC
+// - or ([]emptySet, error) in case the operator or the key is missing in the matchExpressions Selector part of the PVC
+// Example:
+// selector:
+//     matchExpressions:
+//       - key: failure-domain.beta.kubernetes.io/zone
+//         operator: In
+//         values:
+//           - us-east-1a
+//           - us-east-2a
+//           - us-east-3a
+//       - key: failure-domain.beta.kubernetes.io/zone
+//             operator: In
+//             values:
+//               - us-east-3a
+//               - us-east-4a
+// Returns (sets.String{"us-east-3a": sets.Empty{}}, nil) because all operators are ANDed.
+func GetPVCMatchExpression(pvc *v1.PersistentVolumeClaim, key string, operator metav1.LabelSelectorOperator) ([]sets.String, error) {
+	if pvc.Spec.Selector == nil {
+		return make([]sets.String, 0), fmt.Errorf("missing selector.matchExpressions")
+	}
+	if len(pvc.Spec.Selector.MatchExpressions) < 1 {
+		return make([]sets.String, 0), fmt.Errorf("key(s), operator(s) and value(s) are missing in selector.matchExpressions")
+	}
+	capacity := 0
+	for _, item := range pvc.Spec.Selector.MatchExpressions {
+		if item.Key == key && item.Operator == operator && len(item.Values) > 0 {
+			capacity++
+		}
+	}
+	if capacity == 0 {
+		return make([]sets.String, 0), fmt.Errorf("operator %q for key %q not found in selector.matchExpressions", key, operator)
+	}
+
+	ret := make([]sets.String, 0, capacity)
+	index := 0
+	for _, item := range pvc.Spec.Selector.MatchExpressions {
+		if item.Key == key && item.Operator == operator && len(item.Values) > 0 {
+			ret = append(ret, make(sets.String))
+			for _, value := range item.Values {
+				ret[index].Insert(value)
+			}
+			index++
+		}
+	}
+	return ret, nil
+}
+
+// RegionsToZones converts a set of regions to a set of zones
```

Comment:
Well, in case an admin doesn't configure any zone(s) in the StorageClass "the default" is a set of all zones that's why the current implementation of the `func GetAllZones` is useful from my point of view.
So I would personally leave the `func GetAllZones` as is.
However, it may be useful to create map[region]SetOfZones data structure and use it for Regions -> SetOfZones conversion instead using the current implementation of the `func RegionsToZones`.